libsolidity/formal/CHC.cpp

```diff
@@ -323,6 +376,109 @@ void CHC::unknownFunctionCall(FunctionCall const&)
 	m_unknownFunctionCallSeen = true;
 }
 
+/*
+ * Loop encoding is described as follows.
+ * Let f_i the function body block before the loop.
+ * Create the following blocks:
+ * loop_header, containing constraints created by the condition (potential side effect statements).
+ * loop_body, containing constraints from the loop body.
+ * f_j, the function body block after the loop.
+ * Create the following edges:
+ * f_i -> loop_header
+ * loop_header -> loop_body
+ * loop_header -> f_j
+ * loop_body -> loop_header, if there were no nested loops inside this loop body.
+ * f_k -> loop_header, if there was at least one nested loop inside this loop body,
```

Comment:
Why would it jump directly to the header? Also, does this implement break and continue?

Reply:
Where else would it jump? `f_k` is the latest function block block created inside the loop body. If it exists, it is created as the exit block of an inner loop. Therefore it needs to go back to `loop_header`.
`break` and `continue` are unsupported atm